### ISSUE TITLE
If you have raptor strike activated to strike on next hit and yo...

### ISSUE THREAD

**Issue by kaytotes** [Project-Epoch/bugtracker#3710]
> If you have raptor strike activated to strike on next hit and you move out of the mobs melee hitbox (only for a fraction of a second) your character will attempt to shoot as ranged instead canceling your raptor strike. 

This should not happen and it crazy annoying to play around with

.go xyz 2047.770000 1795.190000 98.690200 0 1.825380

**Comment by zana244** [comment]
As far as I am aware, on vanilla and TBC client (1.12 and 2.4.3) you can queue raptor strike at any range, and it will never unqueue unless you manage to hit with it. (tested on everlook/stormforge)

However, on the WotLK client, you can only queue raptor strike while in range. Even so, the implementation on internal Epoch is extremely clunky -- even if you are in range, if you queue the attack and you can't instantly hit it (if you are not facing the target), it unqueues itself instantly.

Trying it on the 3.3.5 warmane client, the attack stays queued for as long as you are in melee range, and unqueues either manually or by outranging the target. --> even this is a major improvement, as you do not have to spam a strike like a madman trying to connect.

I'll mark it as a bug and leave it up to the devs to modify it. It's worth noting that the same behaviour applies to Heroic Strike (or any other queueing hit...if there are any others...Maul?) -- the unqueuing in melee range is very clunky, but at least warriors don't need to bother with moving as much as hunters.

Edit: Funnily enough you can queue Heroic Strike from range but not Raptor Strike.
Edit2: There seems to be a position (facing 90 degrees from the target), where the strikes can queue but you cannot attack. The melee clunkiness actually appears if you try to queue while not facing the target -> The spell does not get queued because it fails the facing check "Target needs to be in front of you". If you manage to queue it, you will only get the "You are facing the wrong way" error from auto attack and it will unqueue only when you outrange the enemy.

Looking at the error codes, the major clunk in the hunter implementation is that if you queue Raptor Strike at range you get a flurry of Interrupted errors, most likely due to Auto Shot trying to reactivate constantly (and of course, unqueuing Raptor Strike).

**Comment by zana244** [comment]
Investigated this, asking on the TC discord. With the auto switch toggled on, it seems a HandleAttackStopPacket is being called when moving out of melee range before the Auto Shot spell packet is sent, which threw a wrench in my quest...

<img width="488" height="557" alt="Image" src="https://github.com/user-attachments/assets/6b3e04f3-d47c-4cbd-8b5c-149c881cb54d" />

<img width="498" height="323" alt="Image" src="https://github.com/user-attachments/assets/21b06218-7588-4e97-a056-a8e3130edbba" />

Investigated robinsch's initial idea by toggling the cvar off in a macro with /cast Raptor Strike, and using chatgpt to create an addon to turn the cvar back on upon successful raptor strike or cancelling its cast...

Sadly, if you toggle the cvar back on while you are still attacking, the client does not start sending the auto shot spell packet, even though you are doing the attacking animation / continuing to melee attack if you stay near the mob... you need to manually right click the mob again or manually toggle auto shot back, which defeats the purpose.

Maybe the spellmod fix results in the desired behaviour? will see if I have the motivation to try it but it seems way too hacky --> the cvar would've been purely client side and elegant...

TLDR: If this annoys you, turn the auto switch off and play like in vanilla, or learn to time your raptor strike.
